assets/src/hooks/useShapes.ts

```diff
@@ -30,18 +37,31 @@ export const useRouteShapes = (selectedRouteIds: RouteId[]): Shape[] => {
   useEffect(() => {
     selectedRouteIds.forEach((routeId: RouteId) => {
       if (!(routeId in shapesByRouteId)) {
-        setLoadingShapesForRoute(routeId)
+        if (!isASubwayRoute(routeId)) {
+          setLoadingShapesForRoute(routeId)
 
-        if (isASubwayRoute(routeId)) {
-          setShapesForRoute(routeId, subwayRouteShapes(routeId))
-        } else {
           fetchShapeForRoute(routeId).then((shapes: Shape[]) =>
             setShapesForRoute(routeId, shapes)
           )
+        } else if (stations) {
+          setLoadingShapesForRoute(routeId)
+
+          Promise.all(
+            subwayRoutes[routeId].gtfsRouteIds.map((routeId) =>
+              fetchShapeForRoute(routeId)
+            )
+          ).then((shapesLists) => {
+            setShapesForRoute(
+              routeId,
+              flatten(shapesLists).map((shape) =>
+                enhanceShapeForSubwayRoute(shape, routeId, stations || [])
+              )
+            )
+          })
         }
       }
     })
-  }, [selectedRouteIds, shapesByRouteId])
+  }, [selectedRouteIds, shapesByRouteId, stations])
```

Comment:
question: Hmm, is it weird that `setShapesByRouteId` is not in the dependency list? Given there are no linter errors, It seems like the linter is unsure if `setShapesForRoute`, `setLoadingShapesForRoute` change, dispite  `setShapesByRouteId` being able to change.

Maybe try defineing the funcitons within the `useEffect` and see if the linter then complains about the dependencies list?
https://react.dev/reference/react/useEffect#removing-unnecessary-function-dependencies

Reply:
The linter rule actually has some [special logic](https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L155) around a few specific kinds of values that don't need to be declared as deps, and state setters are among them.

Reply:
TIL! Thanks!